pkg/clusters/addons/kong/builder.go

```diff
@@ -0,0 +1,46 @@
+package kong
+
+// -----------------------------------------------------------------------------
+// Kong Addon - Builder
+// -----------------------------------------------------------------------------
+
+// Builder is a configuration tool for Kong cluster.Addons
+type Builder struct {
+	namespace  string
+	name       string
+	deployArgs []string
+	dbmode     DBMode
+}
+
+// NewBuilder provides a new Builder object for configuring and generating
+// Kong Addon objects which can be deployed to cluster.Clusters
+func NewBuilder() *Builder {
+	builder := &Builder{
+		namespace:  DefaultNamespace,
+		name:       DefaultDeploymentName,
+		deployArgs: defaults(),
+	}
+	return builder.WithDBLess()
+}
+
+// WithDBLess configures the resulting Addon to deploy a DBLESS proxy backend.
+func (b *Builder) WithDBLess() *Builder {
+	b.dbmode = DBLESS
+	return b
+}
+
+// WithPostGreSQL configures the resulting Addon to deploy a PostGreSQL proxy backend.
+func (b *Builder) WithPostGreSQL() *Builder {
```

Comment:
nit: I've never seen anyone writing "postgres" with a capital "G"

Wikipedia says
> PostgreSQL (/ˈpoʊstɡrɛs ˌkjuː ˈɛl/),[13] also known as Postgres [...]

Reply:
:+1: changed in 195d6314183ee67bd6737784a9adc3a7b7a2d3ae